language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/rename/RenameUtil.java

```diff
@@ -125,25 +128,68 @@ private RenameUtil() {
         if (onImportDeclarationNode(context, nodeAtCursor)) {
             return Collections.emptyMap();
         }
+
         if (QNameReferenceUtil.onModulePrefix(context, nodeAtCursor)) {
-            return handleQNameReferenceRename(context, document.get(), nodeAtCursor, newName);
+            return handleQNameReferenceRename(context, document.get(), nodeAtCursor, newName, honorsChangeAnnotations);
         }
         if (onImportPrefixNode(context, nodeAtCursor)) {
-            return handleImportPrefixRename(context, document.get(), nodeAtCursor, newName);
+            return handleImportPrefixRename(context, document.get(), nodeAtCursor, newName, honorsChangeAnnotations);
         }
 
         Map<Module, List<Location>> locationMap = ReferencesUtil.getReferences(context);
-
         Map<String, List<TextEdit>> changes = new HashMap<>();
-        locationMap.forEach((module, locations) ->
-                locations.forEach(location -> {
-                    String uri = ReferencesUtil.getUriFromLocation(module, location);
-                    List<TextEdit> textEdits = changes.computeIfAbsent(uri, k -> new ArrayList<>());
+
+        for (Map.Entry<Module, List<Location>> entry : locationMap.entrySet()) {
+            Module module = entry.getKey();
+            List<Location> locations = entry.getValue();
+            for (Location location : locations) {
+                String uri = ReferencesUtil.getUriFromLocation(module, location);
+                List<TextEdit> textEdits = changes.computeIfAbsent(uri, k -> new ArrayList<>());
+                if (honorsChangeAnnotations && CommonUtil.isKeyword(newName)) {
+                    String escapedNewName = CommonUtil.escapeReservedKeyword(newName);
+                    textEdits.add(new AnnotatedTextEdit(ReferencesUtil.getRange(location),
+                            escapedNewName, RenameChangeAnnotation.QUOTED_KEYWORD.getID()));
+                    textEdits.add(new AnnotatedTextEdit(ReferencesUtil.getRange(location),
+                            newName, RenameChangeAnnotation.UNQUOTED_KEYWORD.getID()));
```

Comment:
Get the following part to a variable and reuse
`AnnotatedTextEdit(ReferencesUtil.getRange(location)`

Reply:
Done